examples/secret_storage/secret_storage.go

```diff
@@ -0,0 +1,572 @@
+	if err != nil {
+		if e, ok := err.(kbchat.Error); !ok || e.Code != kbchat.RevisionErrorCode {
+			// unexpected error
+			return false, result, err
+		}
+
+		// failed put. try get
+		result, err := r.Lookup(teamName, tool)
+		if err != nil {
+			return false, result, err // api call failed
+		}
+		return false, result, nil // failed put. return KVGetResult
+	}
+	return true, result, nil // successul put
+}
+
+// Unreserve a tool for a given day if that day is currently reserved by the given user.
+// Note: if you unreserve a not-added or deleted tool, it will not add the tool.
+// Returns (whether action is successful, most recent get result if applicable, error)
+func (r *RentalBotClient) Unreserve(teamName string, username string, tool string, day string) (ok bool, result keybase1.KVGetResult, err error) {
+	var val map[string]string
+	result, err = r.Lookup(teamName, tool)
+	if err != nil {
+		return false, result, err // api call failed
+	} else if result.EntryValue == "" {
+		val = make(map[string]string)
+	} else {
+		if err := json.Unmarshal([]byte(result.EntryValue), &val); err != nil {
+			return false, result, err
+		}
+	}
+
+	reserver, ok := val[day]
+	if !ok {
+		// a noop because currently not reserved
+		return true, result, nil
+	} else if reserver != username {
+		// failed to put because current reserver is not user
+		return false, result, nil
+	}
+	expectedRevision := result.Revision + 1
+	delete(val, day)
+
+	bytes, err := json.Marshal(val)
+	if err != nil {
+		return false, result, err
+	}
+
+	_, err = r.api.PutEntryWithRevision(teamName, r.namespace, tool, string(bytes), expectedRevision)
+	if err != nil {
+		if e, ok := err.(kbchat.Error); !ok || e.Code != kbchat.RevisionErrorCode {
+			// unexpected error
+			return false, result, err
+		}
+
+		// failed put. try get
+		result, err = r.Lookup(teamName, tool)
+		if err != nil {
+			return false, result, err // api call failed
+		}
+		return false, result, nil // failed put. return KVGetResult
+	}
+	return true, result, nil // successul put
+}
+
+func (r *RentalBotClient) ListTools(teamName string) ([]string, error) {
+	var tools []string
+	res, err := r.api.ListEntryKeys(teamName, r.namespace)
+
+	if err != nil {
+		return tools, err
+	}
+
+	for _, tool := range res.EntryKeys {
+		tools = append(tools, tool.EntryKey)
+	}
+	return tools, nil
+}
+
+func fail(msg string, args ...interface{}) {
+	fmt.Fprintf(os.Stderr, msg+"\n", args...)
+	os.Exit(3)
+}
+
+func basicRentalUsers(rental *RentalBotClient, team string) error {
+	user1 := "Jo"
+	user2 := "Charlie"
+	date1 := "2044-03-12"
+	date2 := "2044-06-12"
+	date3 := "2044-06-13"
+	tool := "laz0rs"
+
+	ok, res, err := rental.Remove(team, tool)
+	fmt.Printf("REMOVE: %v, %+v, %v\n", ok, res, err)
+	if !ok || err != nil {
+		return fmt.Errorf("Unexpected result: %v, %v", ok, err)
+	}
+
+	var tools []string
+	tools, err = rental.ListTools(team)
```

Comment:
You should check these errors and return if `!=nil`

Reply:
you can wrap it in a method call. like...
```golang
tools, err = rental.ListTools(team)
failTestIfErr(err)
```

Reply:
i don't see much gain in refactoring a `failTesIfErr` method since sometimes i'm checking for `ok` and sometimes checking for `!ok`; sometimes checking for `err !=nil` and sometimes checking for `err == nil`; sometimes there's `ok` and `err`, sometimes just `err`. and i'd still need to keep the if block ( `if err = failTestIfErr(...); err != nil { return err}`) to have the function return an error, so i wouldn't be reducing number of lines of code